base_layer/core/src/proof_of_work/diff_adj_manager/diff_adj_storage.rs

```diff
@@ -299,8 +306,15 @@ impl DiffAdjStorage {
     // Adds the new PoW sample to the specific LinearWeightedMovingAverage specified by the PoW algorithm.
     fn add(&mut self, timestamp: EpochTime, pow: ProofOfWork) -> Result<(), DiffAdjManagerError> {
         match pow.pow_algo {
-            PowAlgorithm::Monero => self.monero_lwma.add(timestamp, pow.accumulated_monero_difficulty)?,
-            PowAlgorithm::Blake => self.blake_lwma.add(timestamp, pow.accumulated_blake_difficulty)?,
+            PowAlgorithm::Monero => {
+                let target_difficulty = self.monero_lwma.get_difficulty();
+                self.monero_lwma.add(timestamp, target_difficulty)?
```

Comment:
Is this correct, the resulting difficulty of the lwma is add as a sample in the lwma?

Reply:
I think it is. Basically a miner would have called "What is the current target difficulty", which is `lwma.get_difficulty()` (A). Then it would have solved the block according to that target. The block is then added to the LWMA saying that it was solved at target A and at time `header.timestamp`

Reply:
The blocks achieved difficulty should be used and not the get_difficulty of the lwma. These can be different as a remote miner could have solved the block.

Reply:
Random thing, wont the remote target and your local target be different due to different timestamps?

Reply:
I don't think so, because each miner mining at this block height will have calculated the target difficulty the same.